cli/azd/pkg/azdo/azdo.go

```diff
@@ -44,19 +44,35 @@ var (
 	ServiceConnectionName = "azconnection"
 )
 
+type Connection struct {
+	*azuredevops.Connection
+	Organization Organization
+}
+
+type Organization struct {
+	Name string
+}
+
 // helper method to return an Azure DevOps connection used the AzDo go sdk
 func GetConnection(
-	ctx context.Context, organization string, personalAccessToken string) (*azuredevops.Connection, error) {
+	ctx context.Context, organization string, personalAccessToken string) (Connection, error) {
 	if organization == "" {
-		return nil, fmt.Errorf("organization name is required")
+		return Connection{}, fmt.Errorf("organization name is required")
 	}
 
 	if personalAccessToken == "" {
-		return nil, fmt.Errorf("personal access token is required")
+		return Connection{}, fmt.Errorf("personal access token is required")
 	}
```

Comment:
Why are we changing return values? 

For functions that return some type and an error it always feels more appropriate to return a pointer instead of a struct value.  In the error condition return a nil pointer makes it really clear that there is no value for the specified connection and that the error has the reason for why.

Reply:
>  it always feels more appropriate to return a pointer instead of a struct value

I totally agree with this.  But, I've been thinking about this lately and wondering if I am a little bias here for how it looks better for error cases v/s the real benefit of using a struct v/s a pointer for the non-error case.

I'd like to discuss this eventually with the team and hear from others.  I don't think it is such a big deal, but I feel like the decision on when to use struct and when to use *struct should not depend on the error returning case and how it looks, but there should be a better justification for it.